addons/account/models/account_order_mixin.py

```diff
@@ -0,0 +1,249 @@
+    currency_id = fields.Many2one('res.currency', 'Currency', required=True, default=lambda self: self.env.company.currency_id.id)
+    fiscal_position_id = fields.Many2one('account.fiscal.position', string='Fiscal Position', check_company=True)
+    partner_id = fields.Many2one('res.partner', required=True, change_default=True, tracking=True, index=True, domain="[('company_id', 'in', (False, company_id))]")
+
+    amount_untaxed = fields.Monetary(string='Untaxed Amount', store=True, compute='_compute_amounts')
+    amount_tax = fields.Monetary(string="Taxes", store=True, compute='_compute_amounts')
+    amount_total = fields.Monetary(string='Total', store=True, compute='_compute_amounts')
+
+    invoice_ids = fields.Many2many('account.move')
+    order_line = fields.One2many('account.order.line.mixin')
+
+    invoice_status = fields.Selection([], compute='_compute_invoice_status', store=True)
+    name = fields.Char(string="Order Reference", required=True, copy=False, readonly=False, index='trigram', default=lambda self: _('New'))
+    payment_term_id = fields.Many2one('account.payment.term', 'Payment Terms', check_company=True)
+
+    def _sign(self):
+        """
+        Returns the sign (1 or -1) to multiply with when calculating the amount to invoice and creating the invoice lines.
+        """
+        raise NotImplementedError  # To override
+
+    def _is_locked(self):
+        """
+        Returns whether the order is locked.
+        Used to decide which down payment lines to update the status of when posting a linked down payment invoice.
+        """
+        raise NotImplementedError  # To override
+
+    def _filter_subrecords(self):
+        """
+        Returns a filter function that only returns True for invoice lines linked to this order.
+        Used to compute the amount to invoice.
+        """
+        raise NotImplementedError  # To override
+
+    def _create_new_order_line(self, values):
+        """
+        Returns the results of a call to create() on the subrecord (sale.order.line or purchase.order.line).
+        Used to create the order lines of the correct type.
+        """
+        raise NotImplementedError  # To override
+
+    def _prepare_invoice(self):
+        """
+        Prepare the dict of values to create the new invoice for this order.
+        Used when generating the invoice from this account order.
+        """
+        self.ensure_one()
+        return {
+            'currency_id': self.currency_id.id,
+            'invoice_origin': self.name,
+            'invoice_payment_term_id': self.payment_term_id.id,
+            'invoice_line_ids': [],
+            'company_id': self.company_id.id,
+        }
+
+    def _create_invoices(self, grouped=False, final=False, date=None):
+        """
+        Create the invoices for this account order.
+        """
+        raise NotImplementedError  # To override
+
+    def action_view_invoice(self, invoices=False):
```

Comment:
Order the methods in this mixin according to the [Model attribute order](https://www.odoo.com/documentation/17.0/contributing/development/coding_guidelines.html#symbols-and-conventions) standard